pkg/controller/endpointslice/reconciler_test.go

```diff
@@ -177,6 +217,19 @@ func TestReconcile1Pod(t *testing.T) {
 				t.Errorf("Expected EndpointSlice to have label set with %s value, got %s", testCase.service.Name, slice.Labels[discovery.LabelServiceName])
 			}
 
+			// slice inherit parent service labels but reserved ones
+			for k, v := range testCase.service.Labels {
```

Comment:
I think instead of adding all the logic here it might be better to add an `expectedLabels` field to the test cases so there's less of a chance that we'd get the logic wrong in the test.

Reply:
done